sdl_android/src/main/java/com/smartdevicelink/api/audio/AudioStreamManager.java

```diff
@@ -0,0 +1,271 @@
+import com.smartdevicelink.proxy.rpc.AudioPassThruCapabilities;
+import com.smartdevicelink.proxy.rpc.enums.SystemCapabilityType;
+
+import java.io.File;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * The AudioStreamManager class provides methods to start and stop an audio stream
+ * to the connected device. Audio files can be pushed to the manager in order to
+ * play them on the connected device. The manager uses the Android built-in MediaCodec.
+ */
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public class AudioStreamManager extends BaseSubManager {
+    private static final String TAG = AudioStreamManager.class.getSimpleName();
+
+    private IAudioStreamListener sdlAudioStream;
+    private int sdlSampleRate;
+    private @SampleType int sdlSampleType;
+    private final Queue<BaseAudioDecoder> queue;
+
+    private StreamingStateMachine streamingStateMachine;
+
+    // INTERNAL INTERFACE
+
+    private final ISdlServiceListener serviceListener = new ISdlServiceListener() {
+        @Override
+        public void onServiceStarted(SdlSession session, SessionType type, boolean isEncrypted) {
+            if (SessionType.PCM.equals(type)) {
+                sdlAudioStream = session.startAudioStream();
+                streamingStateMachine.transitionToState(StreamingStateMachine.STARTED);
+            }
+        }
+
+        @Override
+        public void onServiceEnded(SdlSession session, SessionType type) {
+            if (SessionType.PCM.equals(type)) {
+                if (internalInterface != null) {
+                    session.stopAudioStream();
+                    sdlAudioStream = null;
+                    internalInterface.removeServiceListener(SessionType.PCM, this);
+                    streamingStateMachine.transitionToState(StreamingStateMachine.NONE);
+                }
+            }
+        }
+
+        @Override
+        public void onServiceError(SdlSession session, SessionType type, String reason) {
+            Log.e(TAG, "OnServiceError: " + reason);
+        }
+    };
+
+    /**
+     * Creates a new object of AudioStreamManager
+     * @param internalInterface The internal interface to the connected device.
+     */
+    public AudioStreamManager(@NonNull ISdl internalInterface) {
+        super(internalInterface);
+        this.queue = new LinkedList<>();
+
+        streamingStateMachine = new StreamingStateMachine();
+        transitionToState(BaseSubManager.READY);
+    }
+
+    @Override
+    public void dispose() {
+        stopAudioStream();
+
+        streamingStateMachine.transitionToState(StreamingStateMachine.NONE);
+    }
+
+    /**
+     * Starts the audio service and audio stream to the connected device.
+     * The method is non-blocking.
+     * @param encrypted Specify whether or not the audio stream should be encrypted.
+     */
+    public void startAudioStream(boolean encrypted) {

```

Comment:
there should be a callback on this. Currently if theres an issue, the developer would be none the wiser, and it would negatively impact the rest of the session. Even if its the simple `CompletionListener` that we are using everywhere, it would be better it currently is